src/components/PostDetails.tsx

```diff
@@ -1,107 +1,125 @@
+        setComments(prev => [...prev, comment]),
+      );
+    },
+    [postId],
+  );
 
-          <div className="notification is-danger" data-cy="CommentsError">
-            Something went wrong
-          </div>
+  const onDeleteComment = (id: number) => {
+    removeComment(id);
+    setComments(prev => prev.filter(item => item.id !== id));
+  };
 
-          <p className="title is-4" data-cy="NoCommentsMessage">
-            No comments yet
-          </p>
+  useEffect(() => {
+    setComments([]);
+    setIsAddFormActive(false);
+  }, [postId]);
 
-          <p className="title is-4">Comments:</p>
+  useEffect(() => {
+    setErrorComment(false);
+    setIsCommentsLoading(true);
+    getCommentsByPostId(postId)
+      .then(setComments)
+      .catch(() => setErrorComment(true))
+      .finally(() => setIsCommentsLoading(false));
+  }, [postId]);
 
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
-            </div>
+  return (
+    <div className="content" data-cy="PostDetails">
+      <div className="content" data-cy="PostDetails">
+        <div className="block">
+          <h2 data-cy="PostTitle">{`#${postId}: ${title}`}</h2>
 
-            <div className="message-body" data-cy="CommentBody">
-              Some comment
-            </div>
-          </article>
+          <p data-cy="PostBody">{body}</p>
+        </div>
 
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
+        <div className="block">
+          {isCommentsLoading && <Loader />}
 
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
-            </div>
-            <div className="message-body" data-cy="CommentBody">
-              One more comment
+          {errorComment && (
+            <div className="notification is-danger" data-cy="CommentsError">
+              Something went wrong
             </div>
-          </article>
+          )}
 
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
+          {comments.length > 0 && !isCommentsLoading && (
```

Comment:
```suggestion
          {!!comments.length && !isCommentsLoading && (
```
**PLS DO THE SAME FOR ALL SUCH CASES IN THE PROJECT**